Applications/WebPortal/WebApp/library/source/pages/account_directory_page/local_account_directory_model.ts

```diff
@@ -0,0 +1,51 @@
+import * as Beam from 'beam';
+import { AccountDirectoryModel } from './account_directory_model';
+import { AccountEntry } from './account_entry';
+
+/** Implements an AccountDirectoryModel in memory. */
+export class LocalAccountDirectoryModel extends AccountDirectoryModel {
+
+  /** Constructs an new model.
+   * @param groups - A set of groups.
+   * @param accounts - A map of all the accounts associated with
+   *        each group.
+   */
+  constructor(groups: Beam.Set<Beam.DirectoryEntry>,
+      accounts: Beam.Map<Beam.DirectoryEntry, AccountEntry[]>) {
+    super();
+    this._isLoaded = false;
+    this._groups = groups.clone();
+    this._accounts = new Beam.Map<Beam.DirectoryEntry, AccountEntry[]>();
+    for(const group of groups) {
+      this._accounts.set(group, accounts.get(group));
```

Comment:
Close but not good enough, a mean person can still mess things up, but how much they can mess up has been significantly reduced.